CHANGELOG

```diff
@@ -1,3 +1,336 @@
+post-processing, or to better fit with existing application design.
+
+Two forms of lookaside processing are provided: the `odp_ipsec_in()` and
+`odp_ipsec_out()` APIs provide synchronous decrypt and encrypt support,
+respectively. The corresponding `odp_ipsec_in_enq()` and `odp_ipsec_out_enq()`
+APIs provide these same services in asynchronous form where operations can be
+launched and completed later.
+
+====== Inline Support
+In contrast to lookaside support, IPsec inline support permits applications to
+fully leverage the offload capabilities found in many SoCs by allowing inbound
+IPsec packets to be recognized and decrypted automatically before they are
+presented to the application for processing. This is done by configuring a
+Security Association (SA) to operate in inline mode.
+
+Similarly, the `odp_ipsec_out_inline()` API permits a packet to be
+encrypted into an IPsec packet and automatically scheduled for TX processing
+in a single operation. Such "fire and forget" processing enables applications
+to leverage IPsec HW support for such processing in a portable manner.
+
+Applications using IPsec inline support need only "touch" a packet once
+compared to three times when using lookaside processing, leading to greater
+processing efficiency.
+
+====== IPsec Events
+New event types and subtypes are introduced to provide support for IPsec
+processing. The `ODP_EVENT_PACKET` type has a new subtype:
+`ODP_EVENT_PACKET_IPSEC` that provides extended metadata associated with
+IPsec packets that have been processed. The new `ODP_EVENT_IPSEC_STATUS`
+event, in turn, is used to report IPsec status events such as completion
+notifications associated with `odp_ipsec_sa_disable()` calls.
+
+The `odp_ipsec_result()` API is used to obtain IPsec result metadata from
+a packet that has event subtype `ODP_EVENT_PACKET_IPSEC`, while the
+`odp_ipsec_status()` API is used to obtain IPsec status metadata from an
+`ODP_EVENT_IPSEC_STATUS` event.
+
+===== Parser APIs
+Packet parsing has been overhauled with the introduction of two new APIs:
+
+* `odp_packet_parse()`
+* `odp_packet_parse_multi()`
+
+These use an `odp_packet_parse_param_t` struct to control the type and depth
+of parsing to be performed. These routines are intended to be used to
+process packets that have been decapsulated following IPsec decryption or other
+tunneling or on IP fragments after they have been reassembled.
+
+Associated with this improved parse support, the `odp_parser_layer_t` struct
+is deprecated and replaced with a more general `odp_proto_layer_t` struct that
+is used both in PktIO configuration as well as the new parser APIs.
+
+===== Crypto AES-CTR and AES-GMAC Support
+The ODP crypto APIs are extended to provide support for AES-CTR cipher and
+AES-GMAC authentication modes, reflecting the growing availability of
+accelerated support for these.
+
+===== Crypto removal of DES-CBC
+DES-CBC is no longer considered secure and support for it is removed in ODP.
```

Comment:
From linux-generic. Algorithm definition still exists for those who really want it. Maybe we should deprecate DES-CBC alltogether?

Reply:
DES-CBC is obsolete. I thought we had simply removed it. No modern application should be using it and I doubt if legacy apps would object to updating to a more modern cipher if they wished to move to ODP.

Reply:
I'll send a patch then...

Reply:
What if an application needs DES-CBC e.g. for communicating with another device (e.g. some legacy network equipment or storage device)? It is not clear that removing DEC-CBC altogether from ODP crypto buys much.

Reply:
@JannePeltonen well. Single-key DES was deprecated since 1999 (with FIPS 46-3 stating that DES is to be used only for legacy systems) and was withdrawn alltogether (as FIPS standard) in 2005. If somebody really needs single-key DES to talk with legacy system, he can use 3DES with the key repeated 3 times. It will be slow, but we are talking about legacy systems, so speed should not really matter that much.

Reply:
@lumag Good point, especially since a 2018 system should be able to do 3DES faster than a 2005 system could do DES :)